app/models/manageiq/providers/redhat/inventory/collector.rb

```diff
@@ -20,13 +20,13 @@ def initialize(manager, _target)
   end
 
   def initialize_inventory_sources
-    @ems_clusters   = []
-    @networks       = []
-    @storagedomains = []
-    @datacenters    = []
-    @hosts          = []
-    @vms            = []
-    @templates      = []
+    @ems_clusters   = nil
+    @networks       = nil
+    @storagedomains = nil
+    @datacenters    = nil
+    @hosts          = nil
+    @vms            = nil
+    @templates      = nil
```

Comment:
I think we should do this going forward but since we want to backport this lets not introduce too much change

Reply:
ok